### ISSUE TITLE
Option to allow any element as self-closing

### ISSUE THREAD

**Issue by sergey-v9** [yeonjuan/html-eslint#99]
I am trying to adopt your linter in our project to validate small HTML-like template files and found that it has no option to allow any element to be self-closing.
I khow it is a non-standard HTML syntax (more like an [XML Empty-element tags](https://www.w3.org/TR/xml/#d0e2480)), but this is what we have in our templates.

Would you mind adding a way to solve this? Maybe another option for the `require-closing-tags` rule in addition to `selfClosing`? Or do you prefer to stick with HTML standard syntax?

[playground example ](https://yeonjuan.github.io/html-eslint/playground#eyJjb2RlIjoiPGRpdj5cbiAgICA8Y3VzdG9tLXRhZyAvPlxuICAgIDxkaXY+XG4gICAgPC9kaXY+XG48L2Rpdj5cbiIsInJ1bGVzIjp7IkBodG1sLWVzbGludC9pbmRlbnQiOiJlcnJvciIsIkBodG1sLWVzbGludC9yZXF1aXJlLWxpLWNvbnRhaW5lciI6Im9mZiIsIkBodG1sLWVzbGludC9uby1leHRyYS1zcGFjaW5nLWF0dHJzIjoib2ZmIiwiQGh0bWwtZXNsaW50L3JlcXVpcmUtZG9jdHlwZSI6Im9mZiIsIkBodG1sLWVzbGludC9uby1kdXBsaWNhdGUtaWQiOiJvZmYiLCJAaHRtbC1lc2xpbnQvbm8taW5saW5lLXN0eWxlcyI6Im9mZiIsIkBodG1sLWVzbGludC9uby1vYnNvbGV0ZS10YWdzIjoib2ZmIiwiQGh0bWwtZXNsaW50L3JlcXVpcmUtY2xvc2luZy10YWdzIjoiZXJyb3IiLCJAaHRtbC1lc2xpbnQvcmVxdWlyZS1tZXRhLWNoYXJzZXQiOiJvZmYiLCJAaHRtbC1lc2xpbnQvbm8tdGFyZ2V0LWJsYW5rIjoib2ZmIiwiQGh0bWwtZXNsaW50L25vLWR1cGxpY2F0ZS1hdHRycyI6Im9mZiIsIkBodG1sLWVzbGludC9yZXF1aXJlLWJ1dHRvbi10eXBlIjoib2ZmIiwiQGh0bWwtZXNsaW50L25vLXJlc3RyaWN0ZWQtYXR0cnMiOiJvZmYiLCJAaHRtbC1lc2xpbnQvcmVxdWlyZS1pbWctYWx0Ijoib2ZmIiwiQGh0bWwtZXNsaW50L25vLWFjY2Vzc2tleS1hdHRycyI6Im9mZiIsIkBodG1sLWVzbGludC9uby1hcmlhLWhpZGRlbi1ib2R5Ijoib2ZmIn19) with warning
```html
<div>
    <custom-tag />
    <div>
    </div>
</div>
```

**Comment by yeonjuan** [comment]
@sergey-v9 Hi, Thanks for the new issue.
I agree about the feature but it's hard to implement it quickly.
Our parser follows standard HTML syntax which does not allow self-closing for custom elements so it's hard to handle at the linter level. 

P.S.
For solving these kinds of problems, I'm developing a new HTML parser that relaxes some restrictions. (https://github.com/yeonjuan/sa-html-parser).
